AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -137,22 +135,37 @@ public class AWSDataStoreObserveQueryOperation<M: Model>: AsynchronousOperation,
         return observeQueryPublisher.eraseToAnyPublisher()
     }
 
+    var isSynced: Bool {
```

Comment:
is there a possibility that this requires being thread safe? same for `currentSnapshot`?